---
title: "Dada2 Processing"
author: "Aleksei Zverev"
date: "2023-05-26"
output: 
  html_document: 
    keep_md: yes
---

```{r setup, include=TRUE, warning = FALSE, message = FALSE}
knitr::opts_chunk$set(fig.width=14, fig.height=8) 

library('dada2')
library('phyloseq')
library('dplyr')



set.seed(5678)
setwd('~/Praktikum/TyumGU_2023/')
```

## Read files and data

```{r}
path <- 'raw'
list.files(path) %>% list()
```

```{r}
metadata <- read.csv('metadata.csv')
metadata$SampleID <- paste(metadata$Soil, metadata$MainPlant, metadata$Spot, metadata$Illumina_run, sep=".")
metadata
```

## Run DADA2 pipeline

A realisation of a basic tutorial from
https://benjjneb.github.io/dada2/tutorial.html

```{r}
# Forward and reverse fastq filenames have format: SAMPLENAME_R1_001.fastq and SAMPLENAME_R2_001.fastq
fnFs <- sort(list.files(path, pattern="_R1_001.fastq", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_001.fastq", full.names = TRUE))
# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
sample.names
```
### Quality plot

```{r, cache=TRUE}
plotQualityProfile(fnFs[1:2])

# !Long Operations
plotQualityProfile(fnFs, aggregate = T)
plotQualityProfile(fnRs, aggregate = T)
```
### Filter and Trim

```{r}
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(240,180),
              maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=TRUE)
out
```

### Trimmed quality plot

```{r cache=TRUE}
# !Long Operations
plotQualityProfile(filtFs, aggregate = T)
plotQualityProfile(filtRs, aggregate = T)
```

Reads are trimmed fairly, everything is OK, go to the next step

### Build a model and apply it

```{r, cache=TRUE}
# !Long Operation
errF <- learnErrors(filtFs, multithread=TRUE)
errR <- learnErrors(filtRs, multithread=TRUE)
plotErrors(errF, nominalQ=TRUE)

dadaFs <- dada(filtFs, err=errF, multithread=TRUE)
dadaRs <- dada(filtRs, err=errR, multithread=TRUE)
```

### Merge reads and create table

```{r}
# !Long Operation
mergers <- mergePairs(dadaFs, filtFs, dadaRs, filtRs, verbose=TRUE)
seqtab <- makeSequenceTable(mergers)
dim(seqtab)
```

### Taxonomy annotation

```{r}
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
dim(seqtab.nochim)
sum(seqtab.nochim)/sum(seqtab)
```

```{r}
getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
# If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
track
```

Merging leads to losses in reads. Re-run more relaxed filtering

```{r cache=TRUE}
taxa <- assignTaxonomy(seqtab.nochim, "~/tax_n_refs/silva_nr_v138_train_set.fa.gz", multithread=TRUE)

taxa.print <- taxa # Removing sequence rownames for display only
rownames(taxa.print) <- NULL
head(taxa.print)
```
 
```{r}
rownames(metadata) <- metadata$Filename

ps <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows=FALSE), 
               sample_data(metadata), 
               tax_table(taxa))
ps
sample_names(ps)
```
 
### Rename phyloseq-object according to our needs
 
```{r}
## Rename Samples
new.names <- ps@sam_data %>% 
  data.frame() %>% 
  dplyr::select(Filename, SampleID) %>%  
  arrange(Filename, levels = sample_names(ps))

if (all(sample_names(ps) == new.names$Filename)) {
  sample_names(ps) <- ps@sam_data$SampleID
  print("Renamed")
}

sample_names(ps)
```


```{r}
dna <- Biostrings::DNAStringSet(taxa_names(ps))
names(dna) <- taxa_names(ps)
ps <- merge_phyloseq(ps, dna)
taxa_names(ps) <- paste0("ASV", seq(ntaxa(ps)))
```

## Save phyloseq-object and aquire checksum

```{r}
saveRDS(ps, "ps.RData")
ps <- readRDS("ps.RData")

tools::md5sum("ps.RData")
```

